Source/MLX/Factory.swift

```diff
@@ -25,7 +25,9 @@ extension MLXArray {
     static public func zeros<T: HasDType>(
         _ shape: [Int], type: T.Type = Float.self, stream: StreamOrDevice = .default
     ) -> MLXArray {
-        MLXArray(mlx_zeros(shape.map { Int32($0) }, shape.count, T.dtype.cmlxDtype, stream.ctx))
+        var result = mlx_array_new()
+        mlx_zeros(&result, shape.map { Int32($0) }, shape.count, T.dtype.cmlxDtype, stream.ctx)
+        return MLXArray(result)
```

Comment:
This is a typical operation:

- create the result (`mlx_array_new()`)
- call the function (`mlx_zeros`)
- return the result, transfer ownership to it (`MLXArray(result)`)